ros-setup/install.bash

```diff
@@ -5,17 +5,17 @@ then
     tue-install-echo "Adding ROS sources to apt-get"
     sudo sh -c 'echo "deb http://packages.ros.org/ros/ubuntu $(lsb_release -sc) main" > /etc/apt/sources.list.d/ros-latest.list'
 
-    sudo apt-key adv --keyserver 'hkp://keyserver.ubuntu.com:80' --recv-key C1CF6E31E6BADE8868B172B4F42ED6FBAB17C654
+    tue-install-pipe sudo apt-key adv --keyserver 'hkp://keyserver.ubuntu.com:80' --recv-key C1CF6E31E6BADE8868B172B4F42ED6FBAB17C654
 
     tue-install-apt-get-update
     tue-install-debug "Added ROS sources to apt-get successfully"
 else
     tue-install-debug "ROS sources already added to apt-get"
 fi
 
-# TEMP fix for to only update the key
-if ! apt-key adv --list-public-keys 2>/dev/null | grep -q AB17C654
+# Check for expired key and update
+if sudo apt-key adv --list-keys C1CF6E31E6BADE8868B172B4F42ED6FBAB17C654 2>/dev/null | grep -q expired
 then
-    sudo apt-key del 421C365BD9FF1F717815A3895523BAEEB01FA116
-    sudo apt-key adv --keyserver 'hkp://keyserver.ubuntu.com:80' --recv-key C1CF6E31E6BADE8868B172B4F42ED6FBAB17C654
+    tue-install-echo "Updating expired GPG key"
+    tue-install-pipe sudo apt-key adv --keyserver 'hkp://keyserver.ubuntu.com:80' --recv-key C1CF6E31E6BADE8868B172B4F42ED6FBAB17C654
```

Comment:
This can be replaced by this: https://github.com/ros2/ros2_documentation/pull/1189/files

To avoid maintaining these keys in the future.

Reply:
Interesting thing is that apt-key is also deprecated. I'm wondering what is the right way of checking if a key has expired using the method described on the PR you mentioned @Timple . Do you suggest to always execute the `curl` command upon running a update?

Reply:
Never mind a found a pretty robust solution. Will open a PR for it.

Reply:
Reminder

Reply:
See #232